PSReadLine/Render.cs

```diff
@@ -218,36 +218,188 @@ private void RenderWithPredictionQueryPaused()
+            if (!string.IsNullOrEmpty(_statusLinePrompt))
+            {
+                buffer.Append("\n");
+                buffer.Append(_statusLinePrompt);
+                buffer.Append(_statusBuffer);
+            }
+
+            string currentBuffer = buffer.ToString();
+            string previousBuffer = _previousRender.lines[0].Line;
+
+            // In case the buffer was resized.
+            RecomputeInitialCoords(isTextBufferUnchanged: false);
+
+            // Make cursor invisible while we're rendering.
+            _console.CursorVisible = false;
+
+            // Calculate what to render and where to start the rendering.
+            int commonPrefixLength = FindCommonPrefixLength(previousBuffer, currentBuffer);
+
+            // If we're scrolling through history we always want to re-render.
+            // Writing only the diff in this scenario is a weird UX.
+            bool renderDiff = commonPrefixLength > 0 && _anyHistoryCommandCount == 0;
+
+            if (currentBuffer == previousBuffer)
+            {
+                // No-op, such as when selecting text or otherwise re-entering.
+            }
+            else if (previousBuffer.Length == 0)
+            {
+                // Previous buffer was empty so we just render the current buffer.
+                _console.SetCursorPosition(_initialX, _initialY);
+                _console.Write(currentBuffer);
+            }
+            else if (renderDiff)
+            {
+                // We need to differentially render, possibly with a partial rewrite.
+                if (commonPrefixLength != previousBuffer.Length)
+                {
+                    // The buffers share a common prefix but the previous buffer has additional content.
+                    // Move cursor to where the difference starts and clear so we can rewrite.
+                    var diffPoint = ConvertOffsetToPoint(commonPrefixLength, buffer);
+                    _console.SetCursorPosition(diffPoint.X, diffPoint.Y);
+                    _console.Write("\x1b[0J");
+                } // Otherwise the previous buffer is a complete prefix and we just write.
+
+                var diffData = currentBuffer.Substring(commonPrefixLength);
+                _console.Write(diffData);
+            }
+            else
+            {
+                // The buffers are completely different so we need to rewrite from the start.
+                _console.SetCursorPosition(_initialX, _initialY);
+                _console.Write("\x1b[0J");
+                _console.Write(currentBuffer);
+            }
+
+            // If we had to wrap to render everything, update _initialY
+            var endPoint = ConvertOffsetToPoint(currentBuffer.Length, buffer);
+            if (endPoint.Y >= bufferHeight)
+            {
+
+                // We had to scroll to render everything, update _initialY.
+                int offset = 1; // Base case to handle zero-indexing.
+                if (endPoint.X == 0)
```

Comment:
As is discussed in the comment thread below: https://github.com/PowerShell/PSReadLine/pull/4854#discussion_r2294304909
```suggestion
                if (endPoint.X == 0 && !currentBuffer.EndsWith("\n"))
```